build

```diff
@@ -72,7 +72,7 @@ elsif ($os eq 'linux') {
         $LIBDIR="$ORIGIN/static-deps/lib-x86-32";
         $CFLAGS="-m32 -I$LIBDIR/include/$MULTILIB_DIR";
         $CONFIGURE_FLAGS = "--build=i686-unknown-linux-gnu";
-        $LDFLAGS="-L$LIBDIR/lib -L$LIBDIR/lib/$MULTILIB_DIR";
+        $LDFLAGS="-m32 -L$LIBDIR/lib -L$LIBDIR/lib/$MULTILIB_DIR";
```

Comment:
also I don't get why you need to change the build options for 32 bit -- this is unsupported, and won't even work since 32 bit dependencies are not provided.

Reply:
I know, I was testing this as a check, and noticed the flag was missing (which prevents building the non-GUI bits there). I know the GTK 32 bits are missing, but it is still be better to be correct there. This makes the build work for all non-GUI bits, which is a good start.

Reply:
I can't see any reason to add this change even with your explanation, and the SO link doesn't seem to be related, it doesn't even mention -m32 a single time. So yes, please remove this change.

Reply:
`-m32` passes on the correct flags to the linker, you can test it with or without in Docker, but I am removing it since it does not change whats gets built at the end of the day :-)

Reply:
And our Orcale friend:
https://docs.oracle.com/cd/E18659_01/html/821-1384/bjapr.html

```
Object files or libraries compiled with -m32 cannot be linked with object files or libraries compiled with-m64.

Modules that are compiled with -m32|-m64 must also be linked with -m32|-m64. For a complete list of compiler options that must be specified at both compile time and at link time, see A.1.2 Compile-Time and Link-Time Options
```

Reply:
Do you understand what I'm writing? -m32 is not a linker flag. It's a compiler flag. It's already in CFLAGS. I think you are confusing something. If it's *your* code that doesn't build correctly with -m32 in CFLAGS -- you should be looking for problems in your makefile, instead of adding hacks to other people's code.

Of course I see that in your log that you're running *COMPILER* $(CC), and you are not passing CFLAGS to it.
If you pass CFLAGS to the compiler -- the problem will be resolved.

Reply:
But the gcc calls the linker, and the compile flags (CFLAGS) are not linking flags (for instance optimisation like `-O3` don't mean anything at linking) ...
Thanks anway!

Reply:
When you pass the compiler flags to $(CC), which results in calling the linker: the compiler will filter out any irrelevant flags before passing them to $(LD), and will convert them to the format compatible with $(LD).
The content of LDFLAGS is supposed to be compatible with calling LD directly, which is what most build systems will do. If you pass -m32 to ld -- you'll see an error/warning about an unknown flag somewhere:

```
% ld -m32
ld: unknown option: -m32
```

Reply:
Fair enough!